src/parser.ts

```diff
@@ -696,6 +776,78 @@ function validateArity(name: string, arity: GroqFunctionArity, count: number) {
   }
 }
 
+function argumentShouldBeSelector(
+  namespace: string,
+  functionName: string,
+  args: NodeTypes.ExprNode[]
+) {
+  return (
+    namespace == 'diff' && args.length == 2 && ['changedAny', 'changedOnly'].includes(functionName)
+  )
+}
+
+// An array of arrays where each internal array consists of traversals to be applied to a base node.
```

Comment:
The rest of this file is littered with comments. I generally prefer avoiding comments with well named functions/variables and properly broken down functions, but since this code is so abstract (and hard to follow without context) I've attempted to provide explanation of why we need to do things, and in some cases, what we're actually doing.